dolphinscheduler-alert/dolphinscheduler-alert-server/src/main/java/org/apache/dolphinscheduler/alert/service/ListenerEventPostService.java

```diff
@@ -0,0 +1,262 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.alert.service;
+
+import org.apache.dolphinscheduler.alert.api.AlertChannel;
+import org.apache.dolphinscheduler.alert.api.AlertData;
+import org.apache.dolphinscheduler.alert.api.AlertInfo;
+import org.apache.dolphinscheduler.alert.api.AlertResult;
+import org.apache.dolphinscheduler.alert.config.AlertConfig;
+import org.apache.dolphinscheduler.alert.plugin.AlertPluginManager;
+import org.apache.dolphinscheduler.common.constants.Constants;
+import org.apache.dolphinscheduler.common.enums.AlertStatus;
+import org.apache.dolphinscheduler.common.enums.AlertType;
+import org.apache.dolphinscheduler.common.enums.WarningType;
+import org.apache.dolphinscheduler.common.lifecycle.ServerLifeCycleManager;
+import org.apache.dolphinscheduler.common.thread.BaseDaemonThread;
+import org.apache.dolphinscheduler.common.thread.ThreadUtils;
+import org.apache.dolphinscheduler.common.utils.JSONUtils;
+import org.apache.dolphinscheduler.dao.entity.AlertPluginInstance;
+import org.apache.dolphinscheduler.dao.entity.AlertSendStatus;
+import org.apache.dolphinscheduler.dao.entity.ListenerEvent;
+import org.apache.dolphinscheduler.dao.entity.event.AbstractListenerEvent;
+import org.apache.dolphinscheduler.dao.entity.event.ProcessDefinitionCreatedListenerEvent;
+import org.apache.dolphinscheduler.dao.entity.event.ProcessDefinitionDeletedListenerEvent;
+import org.apache.dolphinscheduler.dao.entity.event.ProcessDefinitionUpdatedListenerEvent;
+import org.apache.dolphinscheduler.dao.entity.event.ProcessEndListenerEvent;
+import org.apache.dolphinscheduler.dao.entity.event.ProcessFailListenerEvent;
+import org.apache.dolphinscheduler.dao.entity.event.ProcessStartListenerEvent;
+import org.apache.dolphinscheduler.dao.entity.event.ServerDownListenerEvent;
+import org.apache.dolphinscheduler.dao.entity.event.TaskEndListenerEvent;
+import org.apache.dolphinscheduler.dao.entity.event.TaskFailListenerEvent;
+import org.apache.dolphinscheduler.dao.entity.event.TaskStartListenerEvent;
+import org.apache.dolphinscheduler.dao.mapper.AlertPluginInstanceMapper;
+import org.apache.dolphinscheduler.dao.mapper.ListenerEventMapper;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.curator.shaded.com.google.common.collect.Lists;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nullable;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public final class ListenerEventPostService extends BaseDaemonThread implements AutoCloseable {
+
+    private static final int QUERY_ALERT_THRESHOLD = 100;
```

Comment:
Can we make this param configurable in config file so that can adjust this value without recompiling.